### PR TITLE
DS 6.5 button updates

### PR DESCRIPTION
**Pull request by msendlakowski**
## Description
- Remove support for "small", "medium", and "extra-large" buttons.  Small is now the default button size.
- Remove "regular" style for DS6 and deprecate it for DS4.
- Remove "legacy" style from DS6.
- Add "delete" style for secondary buttons.
- Add "btn--truncated" and "btn--fixed-height" to go along with the existing "btn--large-truncated" and "btn--large-fixed-height".
- Update button styles for DS6.5.
- Fix hover/focus state for disabled primary links in DS6

## Context
This is a sync up with the latest DS6.5 styles.

## References
Closes https://github.com/eBay/skin/issues/700

## Screenshots
Screen Shot 2019-08-09 at 5 06 25 PM
Screen Shot 2019-08-09 at 5 06 46 PM
Screen Shot 2019-08-09 at 5 07 00 PM
Screen Shot 2019-08-09 at 5 07 09 PM
Screen Shot 2019-08-09 at 5 07 39 PM
Screen Shot 2019-08-09 at 5 07 53 PM
Screen Shot 2019-08-09 at 5 08 07 PM
Screen Shot 2019-08-09 at 5 08 21 PM
Screen Shot 2019-08-09 at 5 08 27 PM

### CODE REVIEW

File: src/less/button/base/button.less [hunk 1]
```diff
@@ -50,44 +50,6 @@ svg.fake-btn__icon {
     .btn-fixed-height-icon-dimensions;
 }
 
-button.btn--regular,
```
Comment (ianmcburnie) [inline src/less/button/base/button.less]: I know I said we can brave with our breaking changes, but I'm a little bit hesitant about completely removing this regular button from the code right now. The different sizes I'm definitely happy to remove (as discussed offline), but I think this regular button might need to linger around a little longer with a deprecation notice. @seangates what do you think?

File: src/less/button/ds6/button.less [hunk 2]
```diff
@@ -56,86 +95,29 @@ a.fake-btn {
...
-            height: 40px;
-        }
-
-        &-truncated {
-            .button-size-rules(10px, 10px);
-            .btn-truncate;
-
-            height: 40px;
-        }
+    &--truncated {
```
Comment (ianmcburnie) [inline src/less/button/ds6/button.less]: Just a notice to everyone @msendlakowski @seangates @agliga @DylanPiercey:

I'm not a fan of this LESS syntax of nesting modifier names using the ampersand symbol. It means we cannot search the source code for `btn--truncated` or `btn--large-truncated`. It's also a little harder to read, in the case of `btn--large-truncated I have to scan back upwards through the code to stitch together the relationships. I don't need to do this if the full selector name is right there in front of me. I've noticed this syntax creeping in to some of our code lately.

Comment (msendlakowski) [reply src/less/button/ds6/button.less]: Makes sense.  And I've had that "search" problem too.
It seems pretty pervasive in the code base.  I'll refactor the buttons to clean this up and add a backlog issue to address it in the other components later.

File: src/less/button/base/button.less (hunk 1 above)
Comment (msendlakowski) [reply src/less/button/base/button.less]: Is "regular" an actual button style, like "primary".   I don't see it in the DS6.5 style guide at all.

Comment (seangates) [reply src/less/button/base/button.less]: Let's keep it, but mark it as `deprecated`.

**Comment by seangates** [comment]
Approved, pending the `destructive` red button doc updates.

**Comment by msendlakowski** [comment]
I'm going to merge this so QE can start testing.  I can update the docs with feedback from the design team with a future pre-release.

**Pull request merged**